File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix = matrix_string.split("\n")
        for i in range(len(self.matrix)):
            self.matrix[i] = [int(x) for x in self.matrix[i].split(" ")]

    def row(self, index):
        if index > len(self.matrix):
            raise ValueError(f"index {index} out of matrix bounds")
        return self.matrix[index - 1]

    def column(self, index):
        if index > len(self.matrix[0]):
            raise ValueError(f"index {index} out of matrix bounds")
        return [row[index - 1] for row in self.matrix]

File: solutions/csharp/matrix/1/Matrix.cs
public class Matrix
{
  private readonly int[][] _data;
  public Matrix(string input)
  {
    _data = input.Split("\n")
      .Select(line => line.Split(" ")
                                .Select(int.Parse)
                                .ToArray())
      .ToArray();
  }

  public int[] Row(int row)
  {
    return _data[row - 1];
  }

  public int[] Column(int col)
  {
    var result = new List<int>();
    for (var rowNumber = 0; rowNumber < _data.Length; rowNumber++)
    {
      result.Add(_data[rowNumber][col - 1]);
    }
    return result.ToArray();
  }
}